python/desc/sims_truthcatalog/sne_truth.py

```diff
@@ -87,33 +91,128 @@ def __getattr__(self, attr):
+        # for applying acceptance cone cut.
+        df['ang_sep'] = angularSeparation(df['ra'].to_numpy(),
+                                          df['dec'].to_numpy(),
+                                          row.snra_in, row.sndec_in)
+        df = df.query(f'ang_sep <= {fp_radius}')
+
+        # Insert the rows into the variability truth table.
+        values = []
+        for visit, band, mjd in zip(df['obsHistID'], df['filter'],
+                                    df['expMJD']):
+            synth_phot = sp_factory.create(mjd)
+            values.append((row.snid_in, visit, mjd, band,
+                           synth_phot.calcFlux(band)))
+        if len(values) == 0:
+            if (verbose):
+                print("Process {}: no rows found for id {}".format(process_num, row.snid_in))                
+            continue
+
+        with sqlite3.connect(outfile) as out_conn:
+            cursor = out_conn.cursor()
+        
+            if not db_lock.acquire(timeout=60.0):
+                print('Process {} failed to acquire output lock'.format(i_p))
+                exit(1)
+
+            cursor.executemany(f'''INSERT INTO {table_name} VALUES (?,?,?,?,?)''', values)
+            out_conn.commit()
+            db_lock.release()
+        
+        num_rows += len(values)
+        if verbose:
+            print('Process {} inserting {} rows for id {} '.format(process_num,
+                                                                   len(values),
+                                                                   row.snid_in))
+        if max_rows is not None and num_rows > max_rows:
+            break
 
 class SNeTruthWriter:
     '''
     Write Summary and Variable truth tables for SNe.
     '''
-    def __init__(self, outfile, sne_db_file):
+    def __init__(self, outfile, sne_db_file, max_parallel=1, dry_run=False):
         """
         Parameters
         ----------
         outfile: str
             Name of the sqlite3 file to contain the truth tables.
         sne_db_file: str
             The sqlite3 file containing the SNe model parameters.
+        sne_limit: int 
+            Maximum number of SNe to process. Default: no limit
+        dry_run: boolean
+            No actual db write.  Default: False
         """
         self.outfile = outfile
         if os.path.isfile(outfile):
             raise OSError(f'{outfile} already exists.')
         if not os.path.isfile(sne_db_file):
             raise FileNotFoundError(f'{sne_db_file} not found.')
+        self.dry_run = dry_run
+        self.sne_db_file = sne_db_file
+
+        q = 'select galaxy_id, c_in, t0_in, x0_in, x1_in, z_in, snid_in, snra_in, sndec_in from sne_params order by rowid '
+        print('The query: ', q)
         with sqlite3.connect(sne_db_file) as conn:
-            self.sne_df = pd.read_sql('select * from sne_params', conn)
+            #self.sne_df = pd.read_sql('select * from sne_params', conn)
```

Comment:
Might as well delete the commented out line.

Reply:
done.